public/lessons/greedy/exchange-arguments.mdx

```diff
@@ -0,0 +1,47 @@
+# Exchange Argument
+An exchange argument is a technique used to prove the optimality of greedy algorithms. It involves showing that any non-optimal solution can be transformed into an optimal one by exchanging elements in a strategic manner.
+
+### Steps to Apply Exchange Arguments
+1. Identify Greedy Choices: Identify the key decisions made by the greedy algorithm at each step.
+
+2. Define Exchange Strategy: Define a strategy for exchanging elements to transform any suboptimal solution into an optimal one.
+
+3. Prove Optimality: Use mathematical induction or contradiction to prove that the exchange strategy always leads to an optimal solution.
+
+## Example Problem
+Let's consider a classic example where exchange arguments can be applied: the Activity Selection Problem.
+
+### Activity Selection Problem
+Given a set of activities each marked by a start time and finish time, the goal is to select the maximum number of non-overlapping activities.
+
+### Greedy Algorithm
+1. Sort activities by finish time.
+
+2. Select the first activity.
```

Comment:
I would add something stating "which corresponds to the activity with the earliest finish time" to help tie everything together.

Reply:
ah fair